src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -53,16 +53,33 @@ public void storeMessage(Message message) {
    * @return a list of messages posted by the user, or empty list if user has never posted a
    *     message. List is sorted by time descending.
    */
-  public List<Message> getMessages(String user) {
+  public List<Message> getMessages(String recipient) {
     List<Message> messages = new ArrayList<>();
 
     Query query =
         new Query("Message")
-            .setFilter(new Query.FilterPredicate("user", FilterOperator.EQUAL, user))
+            .setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient))
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
-
-    return getMessagesHelper(results);
```

Comment:
If you look at the code in the getMessagesHelper() helper function, I think you'll find that it already does everything we need. So there's no need to repeat that code here--we can just call the method as-is.

Reply:
So I should delete the getMessagesHelper() function and its code?

Reply:
Nope, actually the opposite! Since the code we want already exists in getMessagesHelper(), and that method is used in both this getMessages(recipient) and also the getAllMessages() function below, we can just call it with the `return getMessagesHelper(results);` line that was already here in order to do exactly the same thing. This lets us reuse the same code from two places, so we don't have to repeat it and it has a clear name.

Let me know if you want to discuss how method calls like this work!